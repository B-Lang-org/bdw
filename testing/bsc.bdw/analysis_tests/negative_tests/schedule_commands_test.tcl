
# Opens the specified project
handlers::open_project project.bspec 

# Cleans the build results
commands::full_clean

# Compiles the current project
handlers::compile

# Waits until the process with the 'BSPEC(BUILDPID)' id has finished
while {$BSPEC(BUILDPID) != ""} {
        update
}

# Loads the module and shows in the Schedule Analysis window
handlers::show_schedule mkMult4Tbbb

# Shows the rule execution order for the loaded moldule
handlers::get_execution_order

# Shows the rule execution order for the loaded moldule
handlers::get_execution_order mkMultvv2

# Shows the rule execution order for the loaded moldule
handlers::get_rule_info

# Loads the specified module
handlers::load_module mkMult1

# Loads the module and shows in the Schedule Analysis window
handlers::show_schedule mkMult4Tbbb

# Shows the rule execution order for the loaded moldule
handlers::get_execution_order

# Shows the rule execution order for the loaded moldule
handlers::get_execution_order mkMultvv2

# Shows the rule execution order for the loaded moldule
handlers::get_rule_info

# Shows the rule execution order for the loaded moldule
handlers::get_rule_info sss

# Shows the rule execution order for the loaded moldule
handlers::get_rule_relations 

# Shows the rule execution order for the loaded moldule
handlers::get_rule_relations result

# Shows the rule execution order for the loaded moldule
handlers::get_rule_relations resuleeet

# Shows the rule execution order for the loaded moldule
handlers::get_rule_relations "start result ddd" RL_cycle

# Shows the rule execution order for the loaded moldule
handlers::get_rule_relations "start result" "RL_cycle asss"

# Cleans the build results
commands::full_clean

# Closes the current project
handlers::close_project

# Exits the workstation
exit

