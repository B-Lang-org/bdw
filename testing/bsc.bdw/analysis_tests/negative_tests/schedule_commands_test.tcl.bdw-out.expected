Executing: bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries -g mkMult1  mult4/Mult1.bsv 
WARNING: + bsc -u -sim -simdir mult4 -bdir mult4 -info-dir . -keep-fires -p .:%/Libraries -g mkMult1 mult4/Mult1.bsv
RESULT: checking package dependencies
RESULT: compiling mult4/Mult.bsv
RESULT: compiling mult4/Mult1.bsv
RESULT: code generation for mkMult1 starts
WARNING: Warning: "mult4/Mult1.bsv", line 8, column 8: (G0117)
WARNING:   Rule `start' shadows the effects of `cycle' when they execute in the same
WARNING:   clock cycle. Affected method calls:
WARNING:     product.write, mcand.write, mplr.write
WARNING:   To silence this warning, use the `-no-warn-action-shadowing' flag.
RESULT: Elaborated module file created: mult4/mkMult1.ba
RESULT: All packages are up to date.
The module 'mkMult4Tbbb' does not exist.
There is no module loaded.
The module 'mkMultvv2' does not exist.
There is no module loaded.
mkMult1
The module 'mkMult4Tbbb' does not exist.
{{mult4/Mult1.bsv 8 8} G0117 {Warning: "mult4/Mult1.bsv", line 8, column 8: (G0117)
  Rule `start' shadows the effects of `cycle' when they execute in the same
  clock cycle. Affected method calls:
    product.write, mcand.write, mplr.write
  To silence this warning, use the `-no-warn-action-shadowing' flag.
}}
result RL_cycle start
result RL_cycle start
The module 'mkMultvv2' does not exist.
The rule is not specified
Incorrect rule name "sss"
The rules are not specified
The second rule is not specified
The second rule is not specified
Incorrect rule name "ddd"
Incorrect rule name "asss"
